exec/plugin-commands-script-runners/src/buildCommandNotFoundHint.ts

```diff
@@ -1,12 +1,41 @@
 import { type PackageScripts } from '@pnpm/types'
 import didYouMean, { ReturnTypeEnums } from 'didyoumean2'
+import { readdirSync } from 'fs'
+import path from 'path'
+
+export function getNearest (name: string, list?: readonly string[]) {
+  return list && didYouMean(name, list ?? [], {
+    returnType: ReturnTypeEnums.FIRST_CLOSEST_MATCH,
+  })
+}
+
+export function getNearestProgram (opts: {
+  programName: string,
+  dir: string,
+  workspaceDir: string | undefined,
+}) {
+  try {
+    const { programName, dir, workspaceDir } = opts
+    const binDir = path.join(dir, 'node_modules', '.bin')
+    const programList = readdirSync(binDir)
+    if (workspaceDir && workspaceDir !== dir) {
+      const workspaceBinDir = path.join(workspaceDir, 'node_modules', '.bin')
+      programList.push(...readdirSync(workspaceBinDir))
```

Comment:
it is not as easy as reading all files from the `.bin` directory. On Windows the same command will be duplicated a few time with different extensions: `.exe`, `.cmd`, maybe even `.ps1`. So we need to remove the extensions.

Reply:
Can the `didYouMean` function handle duplication, especially when we only need 1 result (the top result)?

Reply:
I don't know if it can handle duplication. I would deduplicated before passing it the array

Reply:
It handles duplication just fine
![Screenshot from 2023-08-03 20-14-59](https://github.com/pnpm/pnpm/assets/11488886/99574a48-3467-4bb0-b077-49b33ab53759)